### DISCUSSION TITLE
post/interning-in-cpython

### DISCUSSION THREAD

**Discussion by giscus** [luminousmen/luminousmen.com#133] [General]
# post/interning-in-cpython

Interning is a special CPython optimization attempt that makes newcomers go crazy. It is a very advanced concept that can speed up programs if used wisely

https://luminousmen.com/post/interning-in-cpython

**Comment by tanuj10sk** [comment]
loved the explanation

**Comment by piregler** [comment]
Hi, the part regarding integer interning is at least in python 3.12 not true.
In python 3.10.13 und 3.12.7 this is true too
x = 150000
y = 150000
print(x is y)  # True

**Reply by jarun-title** [reply]
did you run in .py file? I found that it return True in .py but False I'm .ipynb

**Reply by jarun-title** [reply]
you will also get False if you run .py in REPL or run it line by line. I think python create new object for y if each line of the code was compiled separately of something like that
